app/views/dashboard/_new_hosts_widget.html.erb

```diff
@@ -0,0 +1,11 @@
+<% hosts = @data.hosts.includes(:operatingsystem).preload(:owner).order(:created_at => :desc).limit(8) %>
```

Comment:
Yes, rails joins seem to be broken for polymorphic associations. You can try that yourself, just use `includes(:operatingsystem, :owner)` and add a filter that doesn't match any host.